controller_names:
  - tmr_arm_controller
  - robotiq_gripper_controller

tmr_arm_controller:
  action_ns: follow_joint_trajectory
  type: FollowJointTrajectory
  default: true
  joints:
    - joint_1
    - joint_2
    - joint_3
    - joint_4
    - joint_5
    - joint_6
  constraints:
    stopped_velocity_tolerance: 0.05
    goal_time: 0.0

robotiq_gripper_controller:
  action_ns: gripper_cmd
  type: GripperCommand  # Change to appropriate controller type if needed
  default: true
  joints:
    - robotiq_85_left_knuckle_joint
  constraints:
    stopped_velocity_tolerance: 0.05
    goal_time: 0.0

# controller_names:
#   - tmr_arm_controller
#   - robotiq_gripper_controller

# tmr_arm_controller:
#   type: joint_trajectory_controller/JointTrajectoryController
#   action_name: follow_joint_trajectory
#   joints: 
#     - joint_1
#     - joint_2 
#     - joint_3
#     - joint_4
#     - joint_5
#     - joint_6
#   command_interfaces: [position]            # ★ 新增
#   state_interfaces: [position, velocity]    # ★ 新增
#   constraints:
#     goal_time: 0.0
#     stopped_velocity_tolerance: 0.05

# robotiq_gripper_controller:
#   type: gripper_controllers/GripperCommandController
#   action_name: gripper_cmd
#   action_monitor_rate: 20
#   joints: 
#     - robotiq_85_left_knuckle_joint
#     - robotiq_85_right_knuckle_joint
#   command_interfaces: [position]
#   state_interfaces: [position, velocity]
#   allow_partial_joints_goal: true
#   gains:
#     robotiq_85_left_knuckle_joint: {p: 100.0, d: 1.0, i: 1.0, i_clamp: 1.0}

